va/validation-authority.go

```diff
@@ -71,22 +72,29 @@ func (va ValidationAuthorityImpl) validateSimpleHTTP(identifier core.AcmeIdentif
 			Type:   core.MalformedProblem,
 			Detail: "Identifier type for SimpleHTTP was not DNS",
 		}
+
+		va.log.Debug(fmt.Sprintf("SimpleHTTP [%s] Identifier failure", identifier))
 		return challenge, challenge.Error
 	}
 	hostName := identifier.Value
 
 	// Check for DNSSEC failures for A/AAAA records
 	_, _, err := va.DNSResolver.LookupHost(hostName)
-	if dnssecErr, ok := err.(core.DNSSECError); ok {
-		challenge.Error = &core.ProblemDetails{
-			Type:   core.DNSSECProblem,
-			Detail: dnssecErr.Error(),
-		}
-	} else {
-		challenge.Error = &core.ProblemDetails{
-			Type:   core.ServerInternalProblem,
-			Detail: "Unable to communicate with DNS server",
+	if err != nil {
+		if dnssecErr, ok := err.(core.DNSSECError); ok {
+			challenge.Error = &core.ProblemDetails{
+				Type:   core.DNSSECProblem,
+				Detail: dnssecErr.Error(),
+			}
+		} else {
+			challenge.Error = &core.ProblemDetails{
```

Comment:
I was concerned about that too, but not according to the unit tests. See [dns_test.go:97](https://github.com/letsencrypt/boulder/pull/384/files#diff-e2747dd7e96960b6c77609940ab683f3R97) - nonexistent domains don't return err, they just return an empty set of IPs.